contracts/stake-cw721/src/state.rs

```diff
@@ -0,0 +1,45 @@
+use cosmwasm_std::{Addr, Uint128};
+use cw721_controllers::NftClaims;
+use cw_controllers::{Claims, Hooks};
+use cw_storage_plus::{Item, SnapshotItem, SnapshotMap, Strategy};
+use cw_utils::Duration;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+pub struct Config {
+    pub owner: Option<Addr>,
+    pub manager: Option<Addr>,
+    pub nft_address: Addr,
+    pub reward_token_address: Option<Addr>,
+    pub unstaking_duration: Option<Duration>,
+}
+
+pub const CONFIG: Item<Config> = Item::new("config");
+
+pub const STAKED_NFTS_PER_OWNER: SnapshotMap<&Addr, HashSet<String>> = SnapshotMap::new(
+    "staked_nfts_per_owner",
+    "staked_nfts_per_owner__checkpoints",
+    "staked_nfts_per_owner__changelog",
+    Strategy::EveryBlock,
+);
+
+pub const TOTAL_STAKED_NFTS: SnapshotItem<HashSet<String>> = SnapshotItem::new(
```

Comment:
From my read it seems like we only ever use the length of this `HashSet`. How about just storing a `u64` here?

Reply:
Agreed! How about adding a new query to fetch all the staked tokens? I think this would be a good query function to have. What's your take on this?

Reply:
> Agreed! How about adding a new query to fetch all the staked tokens? I think this would be a good query function to have. What's your take on this?

Yeah, I like this idea! Makes sense for NFTs. Should have pagination.